traffic_ops/traffic_ops_golang/deliveryservice/deliveryservices.go

```diff
@@ -862,8 +863,8 @@ func updateV31(w http.ResponseWriter, r *http.Request, inf *api.APIInfo, dsV31 *
 		if err != nil {
 			return nil, http.StatusInternalServerError, nil, fmt.Errorf("querying delivery service with sslKeyVersion failed: %s", err)
 		}
-		if ds.CDNName != nil && ds.RoutingName != nil {
-			if sslKeysExist && (oldDetails.OldCdnName != *ds.CDNName || oldDetails.OldRoutingName != *ds.RoutingName) {
+		if (ds.CDNName != nil || ds.CDNID != nil) && ds.RoutingName != nil {
```

Comment:
If one of ds.CDNName or ds.CDNID is nil, this will cause the next line to seg fault